operator/pkg/gateway-api/tlsroute.go

```diff
@@ -59,12 +61,66 @@ func (r *tlsRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 					)
 				}
 			}
+			for _, rule := range hr.Spec.Rules {
+				for _, backend := range rule.BackendRefs {
+					if !helpers.IsServiceImport(backend.BackendObjectReference) {
+						continue
+					}
+					svcImport := &mcsapiv1alpha1.ServiceImport{}
+					if err := r.Client.Get(context.Background(), client.ObjectKey{
+						Namespace: helpers.NamespaceDerefOr(backend.Namespace, hr.Namespace),
+						Name:      string(backend.Name),
+					}, svcImport); err != nil {
+						continue
+					}
+
+					// ServiceImport gateway api support is conditioned by the fact
+					// that an actual Service backs it. Other implementations of MCS API
+					// are not supported.
+					svcName, ok := svcImport.Annotations[mcsapicontrollers.DerivedServiceAnnotation]
+					if !ok {
+						continue
```

Comment:
It's not required in conformance yet, but this should also set a `ResolvedRefs` Condition with `status: false`, just like on HTTPRoute. (It is in the spec for the general BackendRef type).

Reply:
Hi! maybe I am wrong but I think this should already be covered by:
https://github.com/cilium/cilium/pull/28769/files#diff-f4dc0299a417a4dfdecdc98e36b45e6707506b37497dc233ac5ab71a286c026eR118
&
https://github.com/cilium/cilium/pull/28769/files#diff-b4cf692bdcfc41993b078f013a3a8bbcee32ff2615c36b2df5a90cafaa440045R151

Reply:
Hmm, I thought I saw some code that did this same check for HTTPRoute, but I can't see it here.

This case is when the ServiceImport CRD exists, but the service that expected derived Service does not, right? So, in that case, by the comment above, the ServiceImport is not the correct implementation, and so is now an invalid backend.

If _any_ backend is invalid, we should be setting ResolvedRefs, since that indicates that at least one reference was invalid for some reason. Ideally, we indicate somehow which backend was invalid in the ResolvedRefs `message` field.

So this block:
```go
					if !ok {
						continue
					}
```
Should really do something to record that that specific backend is invalid in a ResolvedRefs Condition.

Sorry, hope that's clearer.

Reply:
I mean I can do that but looking at the code I thought that everything that were setting a `ResolvedRefs` condition should go into a routecheck. And I believe that `CheckBackendIsExistingServiceImport` does what you are suggesting to add, it checks that the annotation is here and that the service exist:
https://github.com/MrFreezeex/cilium/blob/1b5a911b159317d6603ae6b499e1f25d86a01765/operator/pkg/gateway-api/routechecks/route_checks.go#L222

And it is even tested here so that it behave the same way as a standard non existent service also does: 
https://github.com/MrFreezeex/cilium/blob/1b5a911b159317d6603ae6b499e1f25d86a01765/operator/pkg/gateway-api/httproute_reconcile_test.go#L310 / https://github.com/MrFreezeex/cilium/blob/1b5a911b159317d6603ae6b499e1f25d86a01765/operator/pkg/gateway-api/httproute_reconcile_test.go#L778

Reply:
Okay, thanks for making that clear.